homeassistant/components/generic_hygrostat/humidifier.py

```diff
@@ -0,0 +1,465 @@
+            return
+        self._state = True
+        await self._async_operate(force=True)
+        await self.async_update_ha_state()
+
+    async def async_turn_off(self, **kwargs):
+        """Turn hygrostat off."""
+        if not self._active:
+            return
+        self._state = False
+        if self._is_device_active:
+            await self._async_device_turn_off()
+        await self.async_update_ha_state()
+
+    async def async_set_humidity(self, humidity: int):
+        """Set new target humidity."""
+        if humidity is None:
+            return
+
+        if self._is_away and self._away_fixed:
+            self._saved_target_humidity = humidity
+            await self.async_update_ha_state()
+            return
+
+        self._target_humidity = humidity
+        await self._async_operate(force=True)
+        await self.async_update_ha_state()
+
+    @property
+    def min_humidity(self):
+        """Return the minimum humidity."""
+        if self._min_humidity:
+            return self._min_humidity
+
+        # get default humidity from super class
+        return super().min_humidity
+
+    @property
+    def max_humidity(self):
+        """Return the maximum humidity."""
+        if self._max_humidity:
+            return self._max_humidity
+
+        # Get default humidity from super class
+        return super().max_humidity
+
+    @callback
+    async def _async_sensor_changed(self, entity_id, old_state, new_state):
+        """Handle ambient humidity changes."""
+        if new_state is None:
+            return
+
+        if self._sensor_stale_duration:
+            if self._remove_stale_tracking:
+                self._remove_stale_tracking()
+            self._remove_stale_tracking = async_track_time_interval(
+                self.hass,
+                self._async_sensor_not_responding,
+                self._sensor_stale_duration,
+            )
+
+        await self._async_update_humidity(new_state.state)
+        await self._async_operate()
+        await self.async_update_ha_state()
+
+    @callback
+    async def _async_sensor_not_responding(self, now=None):
+        """Handle sensor stale event."""
+
+        _LOGGER.debug(
+            "Sensor has not been updated for %s",
+            now - self.hass.states.get(self._sensor_entity_id).last_updated,
+        )
+        _LOGGER.warning("Sensor is stalled, call the emergency stop")
+        await self._async_update_humidity("Stalled")
+
+    @callback
+    def _async_switch_changed(self, entity_id, old_state, new_state):
+        """Handle humidifier switch state changes."""
+        if new_state is None:
+            return
+        self.async_schedule_update_ha_state()
+
+    async def _async_update_humidity(self, humidity):
+        """Update hygrostat with latest state from sensor."""
+        try:
+            self._cur_humidity = float(humidity)
+        except ValueError as ex:
+            _LOGGER.error("Unable to update from sensor: %s", ex)
```

Comment:
I think this should be only a warning, this is a recoverable situation. In my opinion an error means, that a real life situation (even if that situation is a sensor "error") is not handled by the system. But generic_hygrostat handles sensor "unavailable" "stalled" etc. errors, so this should be only a warning.

Generic_thermostat silently moves on on this situation, I understand that generic_hygrostat directly wants to handle this situation, but emitting an error instead of silence is another extreme in my opinion. Eg. when my system is restarted, generic_hygrostat is the only component that writes errors into the log (until all my sensors are online again).

Reply:
Agree.